iroh/src/magicsock.rs

```diff
@@ -1739,16 +1812,16 @@ enum DiscoBoxError {
 /// [`RelayActor`]: crate::magicsock::RelayActor
 /// [`MagicSock`]: crate::magicsock::MagicSock
 #[derive(Debug)]
-struct RelayDatagramsQueue {
+struct RelayDatagramRecvQueue {
     queue: ConcurrentQueue<RelayRecvDatagram>,
     waker: AtomicWaker,
 }
 
-impl RelayDatagramsQueue {
+impl RelayDatagramRecvQueue {
     /// Creates a new, empty queue with a fixed size bound of 128 items.
     fn new() -> Self {
         Self {
-            queue: ConcurrentQueue::bounded(128),
+            queue: ConcurrentQueue::bounded(512),
```

Comment:
why this number? also the comment is now outdated

Reply:
This is based on the previous per client queues which were this size IIRC.

At 1500 bytes/packet this is 750KiB of buffer.  I think ideally the datagram queues would all have a capacity based on the number of datagram **bytes** queued rather than number of items.  I'm using 1500 bytes/packet to estimate, but with GSO etc it could also be much larger as the max packet size is 64KiB, but being that big is probably not common.

But I think doing buffers capacity on buffered bytes should be a separate effort, and it isn't very high priority right now.